From: Sheng Yang <sheng@linux.intel.com>
Date: Wed, 21 Jan 2009 06:33:00 +0000 (+0800)
Subject: kvm: external module: Fix build for VT-d/AMD IOMMU
X-Git-Url: http://git.kernel.org/?p=virt%2Fkvm%2Fkvm-userspace.git;a=commitdiff_plain;h=f4d1fb9baa37ac54414ea21d910937a33d885e9a;hp=7b2bbc4aa31828a78bd33ae5b6b39fd925c5de7e

kvm: external module: Fix build for VT-d/AMD IOMMU

The vtd.c has renamed to iommu.c, and config option has changed to
CONFIG_IOMMU_API.

Notice now the host kernel before 2.6.29 can't work with VT-d due to API
changed... At least this patch enabled building with host kernel before 2.6.29
with CONFIG_DMAR.

Signed-off-by: Wei Huang <wei.w.huang@intel.com>
Signed-off-by: Sheng Yang <sheng@linux.intel.com>
Signed-off-by: Avi Kivity <avi@redhat.com>
---

diff --git a/kernel/x86/Kbuild b/kernel/x86/Kbuild
index c4723b1..4ef1168 100644
--- a/kernel/x86/Kbuild
+++ b/kernel/x86/Kbuild
@@ -9,8 +9,8 @@ kvm-objs := kvm_main.o x86.o mmu.o x86_emulate.o ../anon_inodes.o irq.o i8259.o
 ifeq ($(EXT_CONFIG_KVM_TRACE),y)
 kvm-objs += kvm_trace.o
 endif
-ifeq ($(CONFIG_DMAR),y)
-kvm-objs += vtd.o
+ifeq ($(CONFIG_IOMMU_API),y)
+kvm-objs += iommu.o
 endif
 kvm-intel-objs := vmx.o vmx-debug.o ../external-module-compat.o
 kvm-amd-objs := svm.o ../external-module-compat.o
